BridgeCareApp/BridgeCareCodeFirst/Models/DeficientData.cs

```diff
@@ -0,0 +1,125 @@
+﻿using BridgeCareCodeFirst.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+
+namespace BridgeCare.Models
+{
+    public class DeficientData
+    {
+        public DeficientOrTargetTab GetDeficient(SimulationResult data, int[] totalYears)
+        {
+            var db = new BridgeCareContext();
+            IQueryable<DeficientResult> deficientOrTargetList = null;
+            DeficientOrTargetTab Results = null;
+
+            string getReport =
+                "SELECT TargetID, Years, TargetMet, IsDeficient " +
+                    " FROM Target_" + data.NetworkId
+                    + "_" + data.SimulationId;
+
+            try
+            {
+                var rawDeficientList = db.Database.SqlQuery<DeficientResult>(getReport).AsQueryable();
+
+                deficientOrTargetList = rawDeficientList.Where(_ => _.IsDeficient == true);
+
+                var dataList = new DeficientOrTarget();
+                var targetAndYear = dataList.DeficientTargetList(deficientOrTargetList);
+                var deficient = new DeficientData();
+                Results = deficient.DeficientInfo(data, targetAndYear, totalYears);
+            }
+            catch (SqlException ex)
```

Comment:
See if there is a way to encapsulate the repetitive exception code.

Reply:
Moved Exception to ThrowError class